conclave-cloud-auction/README.md

```diff
@@ -0,0 +1,285 @@
+fully patched intel sgx cpu on its side.
+Now this report also consists of the code hash which is running inside the enclave.
+
+Now which Action enclave to trust and whom to give the key to is decided by a policy constraint
+which is decided by the client.
+So the KDS generates a key pair, a public and private key.
+KDS gives the public key to the client, and the corresponding private key is given to the enclave when requested.
+So the client can specify which enclave can the KDS give the private key to using what we call as the policy 
+constraints.
+Action enclave is the enclave where all the action happens where our uploaded functions are executed where the 
+private data is accessed inside an enclave.
+
+## Complete Flow of the Conclave Cloud Auction Application
+
+![](ccl-functions.png)
+
+Let's look at the above diagram and try to understand the flow of the application.
+
+#### Step 1:
+So the user wants to run some logic securely and decides to use an enclave for the same.
+The developer writes the code for the user. The developer generates a hash value for this code, the user inspects the code either himself or with the help of some auditor, and makes sure the code does what its expected to do.The user notes down this hash.
+
+#### Step 2:
+The developer requests a public key and attestation report of the KDS from the KDS .
+The developer encrypts the function which is to be uploaded using this public key and uploads it to the conclave cloud platform. This function will sit in a database in the cloud.
+
+#### Step 3:
+The user can now invoke this function specifying the expected hash of the function., this is when the function is provisioned into an action enclave, ie an enclave is initialized , started with this function.
+The platform will encrypt the invocation input parameters and the hash of the function using the same public key received by the KDS.
+
+#### Step 4:
+The action enclave requests the private key from the KDS. The KDS first validates the attestation report from enclave. Enclave also validates Kds s attestation report. Both mutually remote attest each other. KDS also releases the private key to the enclave if it matches the policy constarints.
+The action enclave will decrypt the function, expected function hash and parameters using this private key inside the enclave. Platform provisions this decrypted function inside the JS engine.
+The platform verifies if the hash of the action enclave matches with that provided by the user. If they match then only the function is executed and response sent back to the client. It encrypts the response to be sent to the client using clients public key.
+
+Thats how end to end encryption works.
+
+
+## Building and deploying the demonstration
+Each component must be built and deployed or hosted in order to run the
+demonstration.
+
+### 1. Backend to allow Conclave Functions to access it for storing user databases.
+
```

Comment:
We need to mention that the path to the Java SDK has to be changed on `build.gradle.kts`.